dags/sentinel-2_data_transfer.py

```diff
@@ -38,57 +43,167 @@ def extract_src_key(src_url):
     :param src_url: Full http path of the object
     :return: Object path relative to the base bucket
     """
+
     matches = (re.finditer("/", src_url))
     matches_positions = [match.start() for match in matches]
     start = matches_positions[2] + 1
     return src_url[start:]
 
+def load_stac(s3_hook, stac_json_path):
+    """
+    load the STAC file associated with each scene
+    :param s3_hook: Hook to interact with S3
+    :param stac_json_path: Path to the STAC file on S3
+    :return: Dictionary of the STAC file
+    """
+
+    content_object = s3_hook.get_key(key=extract_src_key(stac_json_path),
+                     bucket_name=default_args['src_bucket_name'])
+    content = content_object.get()['Body'].read().decode('utf-8')
+    data = json.loads(content)
+    return data
+
+def africa_extent():
+    """
+    Load Africa boundaries
+    :return: Shape object of the boundaries
+    """
+
+    boundaries = json.load(open("data/africa-extent.json"))
+    return shape(boundaries["features"][0]["geometry"])
+
+def scene_stac_utm(stac):
+    """
+    Extract the UTM zone from the STAC file
+    :return: UTM zone
+    """
+
+    return stac['properties']['proj:epsg']
+
+def reproject_scene_geom_to_epsg_4326(stac, scene_geometry):
```

Comment:
I just found out about this library, which might be a better way to do this: https://pypi.org/project/mgrs/

It'll only handle the full scene footprint, but will work fine...

Reply:
I'll add this library as part of future improvements - I need to test it first.